src/psij/job_executor_config.py

```diff
@@ -5,16 +5,22 @@
 class JobExecutorConfig(object):
     """An abstract configuration class for :class:`~psij.JobExecutor` instances."""
 
-    def __init__(self, launcher_log_file: Optional[Path] = None):
+    DEFAULT: 'JobExecutorConfig' = None  # type: ignore
```

Comment:
Because the class is not yet a valid reference at that point. See, e.g., https://stackoverflow.com/questions/4369814/class-variables-of-same-type-in-python

Reply:
Thanks!